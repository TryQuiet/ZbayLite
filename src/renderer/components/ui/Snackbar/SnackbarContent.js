import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import { withStyles } from '@material-ui/core/styles'
import IconButton from '@material-ui/core/IconButton'
import CircularProgress from '@material-ui/core/CircularProgress'
import SnackbarContentMaterial from '@material-ui/core/SnackbarContent'

import CloseIcon from '@material-ui/icons/Close'
import CheckCircleIcon from '@material-ui/icons/CheckCircle'
import ErrorIcon from '@material-ui/icons/Error'
import InfoIcon from '@material-ui/icons/Info'
import WarningIcon from '@material-ui/icons/Warning'

import green from '@material-ui/core/colors/green'
import blue from '@material-ui/core/colors/blue'
import amber from '@material-ui/core/colors/amber'

const styles = theme => ({
  close: {
    padding: theme.spacing(0.5),
    margin: 0
  },
  success: {
    backgroundColor: green[600]
  },
  error: {
    backgroundColor: theme.palette.error.dark
  },
  info: {
    backgroundColor: blue[600]
  },
  loading: {
    backgroundColor: blue[400],
    color: '#fff'
  },
  loadingIcon: {
    color: '#fff',
    opacity: 1
  },
  warning: {
    backgroundColor: amber[700]
  },
  fullWidth: {
    'max-width': 'none',
    'flex-grow': 1
  },
  icon: {
    fontSize: 20,
    opacity: 0.9
  },
  content: {
    display: 'flex',
    alignItems: 'center'
  },
  message: {
    paddingLeft: theme.spacing(2)
  }
})

const iconVariants = {
  success: CheckCircleIcon,
  warning: WarningIcon,
  error: ErrorIcon,
  info: InfoIcon,
  loading: CircularProgress
}

export const SnackbarContent = ({
  classes,
  variant,
  message,
  onClose,
  fullWidth
}) => {
  const Icon = iconVariants[variant]
  const closeAction = (
    <IconButton
      key='close'
      color='inherit'
      className={classes.close}
      onClick={onClose}
    >
      <CloseIcon className={classes.icon} />
    </IconButton>
  )
  const action = variant !== 'loading' ? [closeAction] : []
  return (
    <SnackbarContentMaterial
      className={
        classNames({
          [classes[variant]]: true,
          [classes.fullWidth]: fullWidth
        })
      }
      message={
        <span className={classes.content}>
          <Icon className={classNames(classes.icon, classes[`${variant}Icon`])} size={20} />
          <span className={classes.message}>
            {message}
          </span>
        </span>
      }
      action={action}
    />
  )
}

SnackbarContent.propTypes = {
  variant: PropTypes.oneOf(['success', 'warning', 'error', 'info', 'loading']).isRequired,
  message: PropTypes.string.isRequired,
  classes: PropTypes.object.isRequired,
  onClose: PropTypes.func,
  fullWidth: PropTypes.bool
}

SnackbarContent.defaultProps = {
  fullWidth: false
}

export default React.memo(withStyles(styles)(SnackbarContent))
