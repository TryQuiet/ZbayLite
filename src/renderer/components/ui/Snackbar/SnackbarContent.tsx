import React from 'react'
import classNames from 'classnames'

import { makeStyles } from '@material-ui/core/styles'
import IconButton from '@material-ui/core/IconButton'
import CircularProgress from '@material-ui/core/CircularProgress'
import SnackbarContentMaterial from '@material-ui/core/SnackbarContent'

import CloseIcon from '@material-ui/icons/Close'
import CheckCircleIcon from '@material-ui/icons/CheckCircle'
import ErrorIcon from '@material-ui/icons/Error'
import InfoIcon from '@material-ui/icons/Info'
import WarningIcon from '@material-ui/icons/Warning'

import green from '@material-ui/core/colors/green'
import blue from '@material-ui/core/colors/blue'
import amber from '@material-ui/core/colors/amber'

const useStyles = makeStyles(theme => ({
  close: {
    padding: theme.spacing(0.5),
    margin: 0
  },
  success: {
    backgroundColor: green[600]
  },
  error: {
    backgroundColor: theme.palette.error.dark
  },
  info: {
    backgroundColor: blue[600]
  },
  loading: {
    backgroundColor: blue[400],
    color: '#fff'
  },
  loadingIcon: {
    color: '#fff',
    opacity: 1
  },
  warning: {
    backgroundColor: amber[700]
  },
  fullWidth: {
    'max-width': 'none',
    'flex-grow': 1
  },
  icon: {
    fontSize: 20,
    opacity: 0.9
  },
  content: {
    display: 'flex',
    alignItems: 'center'
  },
  message: {
    paddingLeft: theme.spacing(2)
  }
}))

const iconVariants = {
  success: CheckCircleIcon,
  warning: WarningIcon,
  error: ErrorIcon,
  info: InfoIcon,
  loading: CircularProgress
}

interface SnackbarContentProps {
  message: string
  variant: 'success' | 'warning' | 'error' | 'info' | 'loading'
  onClose?: () => void
  fullWidth?: boolean
}

export const SnackbarContent: React.FC<SnackbarContentProps> = ({
  message,
  variant,
  onClose,
  fullWidth = false
}) => {
  const classes = useStyles({})

  const Icon = iconVariants[variant]

  const closeAction = (
    <IconButton key='close' color='inherit' className={classes.close} onClick={onClose}>
      <CloseIcon className={classes.icon} />
    </IconButton>
  )

  const action = variant !== 'loading' ? [closeAction] : []
  const variantIcon = `${variant}Icon` as 'close' | 'content' | 'icon' | 'success' | 'error' | 'info' | 'loading' | 'loadingIcon' | 'warning' | 'fullWidth' | 'message'
  return (
    <SnackbarContentMaterial
      className={classNames({
        [classes[variant]]: true,
        [classes.fullWidth]: fullWidth
      })}
      message={
        <span className={classes.content}>
          <Icon className={classNames(classes.icon, classes[variantIcon])} size={20} />
          <span className={classes.message}>{message}</span>
        </span>
      }
      action={action}
    />
  )
}

export default SnackbarContent
