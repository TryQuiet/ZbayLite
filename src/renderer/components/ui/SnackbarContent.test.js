/* eslint import/first: 0 */
import React from 'react'
import { shallow } from 'enzyme'
import each from 'jest-each'

import { SnackbarContent } from './SnackbarContent'
import { mockClasses } from '../../../shared/testing/mocks'

describe('SnackbarContent', () => {
  each(['success', 'warning', 'error', 'info', 'loading']).test(
    'renders %s',
    (variant) => {
      const result = shallow(
        <SnackbarContent
          variant={variant}
          message='test snackbar'
          onClose={jest.fn()}
          classes={mockClasses}
        />
      )
      expect(result).toMatchSnapshot()
    }
  )

  it('renders fullWidth', () => {
    const result = shallow(
      <SnackbarContent
        variant='success'
        message='test snackbar'
        onClose={jest.fn()}
        classes={mockClasses}
        fullWidth
      />
    )
    expect(result).toMatchSnapshot()
  })
})
