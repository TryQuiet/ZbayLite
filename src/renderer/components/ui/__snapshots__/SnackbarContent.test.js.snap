// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnackbarContent renders error 1`] = `
<WithStyles(ForwardRef(SnackbarContent))
  action={
    Array [
      <WithStyles(ForwardRef(IconButton))
        className="close-mock"
        color="inherit"
        onClick={[MockFunction]}
      >
        <Memo(CloseIcon)
          className="icon-mock"
        />
      </WithStyles(ForwardRef(IconButton))>,
    ]
  }
  className="error-mock"
  message={
    <span
      className="content-mock"
    >
      <Memo(ErrorIcon)
        className="icon-mock errorIcon-mock"
        size={20}
      />
      <span
        className="message-mock"
      >
        test snackbar
      </span>
    </span>
  }
/>
`;

exports[`SnackbarContent renders fullWidth 1`] = `
<WithStyles(ForwardRef(SnackbarContent))
  action={
    Array [
      <WithStyles(ForwardRef(IconButton))
        className="close-mock"
        color="inherit"
        onClick={[MockFunction]}
      >
        <Memo(CloseIcon)
          className="icon-mock"
        />
      </WithStyles(ForwardRef(IconButton))>,
    ]
  }
  className="success-mock fullWidth-mock"
  message={
    <span
      className="content-mock"
    >
      <Memo(CheckCircleIcon)
        className="icon-mock successIcon-mock"
        size={20}
      />
      <span
        className="message-mock"
      >
        test snackbar
      </span>
    </span>
  }
/>
`;

exports[`SnackbarContent renders info 1`] = `
<WithStyles(ForwardRef(SnackbarContent))
  action={
    Array [
      <WithStyles(ForwardRef(IconButton))
        className="close-mock"
        color="inherit"
        onClick={[MockFunction]}
      >
        <Memo(CloseIcon)
          className="icon-mock"
        />
      </WithStyles(ForwardRef(IconButton))>,
    ]
  }
  className="info-mock"
  message={
    <span
      className="content-mock"
    >
      <Memo(InfoIcon)
        className="icon-mock infoIcon-mock"
        size={20}
      />
      <span
        className="message-mock"
      >
        test snackbar
      </span>
    </span>
  }
/>
`;

exports[`SnackbarContent renders loading 1`] = `
<WithStyles(ForwardRef(SnackbarContent))
  action={Array []}
  className="loading-mock"
  message={
    <span
      className="content-mock"
    >
      <WithStyles(ForwardRef(CircularProgress))
        className="icon-mock loadingIcon-mock"
        size={20}
      />
      <span
        className="message-mock"
      >
        test snackbar
      </span>
    </span>
  }
/>
`;

exports[`SnackbarContent renders success 1`] = `
<WithStyles(ForwardRef(SnackbarContent))
  action={
    Array [
      <WithStyles(ForwardRef(IconButton))
        className="close-mock"
        color="inherit"
        onClick={[MockFunction]}
      >
        <Memo(CloseIcon)
          className="icon-mock"
        />
      </WithStyles(ForwardRef(IconButton))>,
    ]
  }
  className="success-mock"
  message={
    <span
      className="content-mock"
    >
      <Memo(CheckCircleIcon)
        className="icon-mock successIcon-mock"
        size={20}
      />
      <span
        className="message-mock"
      >
        test snackbar
      </span>
    </span>
  }
/>
`;

exports[`SnackbarContent renders warning 1`] = `
<WithStyles(ForwardRef(SnackbarContent))
  action={
    Array [
      <WithStyles(ForwardRef(IconButton))
        className="close-mock"
        color="inherit"
        onClick={[MockFunction]}
      >
        <Memo(CloseIcon)
          className="icon-mock"
        />
      </WithStyles(ForwardRef(IconButton))>,
    ]
  }
  className="warning-mock"
  message={
    <span
      className="content-mock"
    >
      <Memo(WarningIcon)
        className="icon-mock warningIcon-mock"
        size={20}
      />
      <span
        className="message-mock"
      >
        test snackbar
      </span>
    </span>
  }
/>
`;
